packages/xpetra/sup/Utils/Xpetra_MatrixMatrix.hpp

```diff
@@ -622,7 +577,13 @@ Note: this class is not in the Xpetra_UseShortNames.hpp
           const Tpetra::CrsMatrix<SC,LO,GO,NO>& tpB = Xpetra::Helpers<SC,LO,GO,NO>::Op2TpetraCrs(B);
           RCP<Tpetra::CrsMatrix<SC,LO,GO,NO> >  tpC = Xpetra::Helpers<SC,LO,GO,NO>::Op2NonConstTpetraCrs(C);
 
-          Tpetra::MatrixMatrix::Add(tpA, transposeA, alpha, tpB, transposeB, beta, tpC);
+          RCP<ParameterList> params = rcp(new ParameterList(""));
+          // params->set("Call fillComplete",false);
+          Tpetra::MatrixMatrix::add(alpha, transposeA, tpA, beta, transposeB, tpB, *tpC,
+                                    Xpetra::toTpetra(transposeA ? A.getRangeMap() : A.getDomainMap()),
```

Comment:
add() always uses the domain and range map of B (or B^T if transposeB) as the domain and range map for C (the documentation is very wrong here, it says it uses the rowmap of B for both). But since you want C to use A's domain/range map here, you could just pass in B before A.